API.md

```diff
@@ -0,0 +1,440 @@
+| `onTap` | `{latitude: 0, longitude: 0, screenCoordX: 0, screenCoordY: 0}` | Fired when the users taps the screen.
+| `onLongPress` | `{latitude: 0, longitude: 0, screenCoordX: 0, screenCoordX: 0}` | Fired when the user taps and holds screen for 1 second.
+| `onStartLoadingMap` | `undefined` | Fired once the map begins loading the style. |
+| `onFinishLoadingMap` | `undefined` | Fired once the map has loaded the style. |
+
+## Methods
+
+You first need to get a ref to your `MapView` component:
+
+```jsx
+<MapView ref={map => { this._map = map; }} />
+```
+
+Then call methods as `this._map.methodName()`.
+
+---
+
+```javascript
+this._map.setDirection(direction, animated = true, callback);
+this._map.setZoomLevel(zoomLevel, animated = true, callback);
+this._map.setCenterCoordinate(latitude, longitude, animated = true, callback);
+this._map.setCenterCoordinateZoomLevel(latitude, longitude, zoomLevel, animated = true, callback);
+this._map.setCenterCoordinateZoomLevelPitch(latitude, longitude, zoomLevel, pitch, animated = true, callback); // Android only
+this._map.setPitch(pitch, animated = true, callback); // Android only
+this._map.easeTo({ latitude, longitude, zoomLevel, direction, pitch }, animated = true, callback);
+```
+
+This set of methods sets the location the map is centered on, the zoom level,
+the heading and the pitch (Android only) of the map.
+
+The transition to the desired location is animated by default, but can be made
+instantaneous by passing `animated` as `false`.
+
+For `easeTo`, all arguments inside the options object are optional. You can specify
+any combination of center coords, zoomLevel, direction and pitch. What is not
+specified stays at their current values.
+
+The methods accept an optional `callback` that will get fired when the animation
+has ended. Additionally, the return value is a promise that gets resolved when the
+animation has ended.
+
+---
+
+```javascript
+this._map.setVisibleCoordinateBounds(latitudeSW, longitudeSW, latitudeNE, longitudeNE, paddingTop = 0, paddingRight = 0, paddingBottom = 0, paddingLeft = 0, animated = true);
+```
+
+This method adjusts the center location and the zoomLevel of the map so that
+the rectangle determined by `latitudeSW`, `longitudeSW`, `latitudeNE`,
+`longitudeNE` fits inside the viewport.
+
+You can optionally pass a minimum padding (in screen points) that will be
+visible around the given coordinate bounds.
+
+The transition is animated unless you pass `animated` as `false`.
+
+---
+
+```javascript
+this._map.setCamera(latitude, longitude, fromDistance, pitch, direction, animationDuration = 0.3); // iOS Only
+```
+
+This method only works on iOS.
+
+Sets the map pitched at an angle (`pitch`, measured in
+degrees, 0 is from straight above), looking towards `latitude` and `longitude`
+from `fromDistance` meters away, pointing towards heading `direction`.
```

Comment:
It’s also possible to create an MGLMapCamera with a viewing distance instead of an altitude. Depending on your use case, you might want one or the other without having to do the trig yourself.